profiling/src/profile/mod.rs

```diff
@@ -364,20 +364,21 @@ impl Profile {
         Ok((label_set_id, timestamp))
     }
 
-    fn extract_api_sample_types(&self) -> Option<Vec<api::ValueType>> {
-        let mut sample_types: Vec<api::ValueType> = Vec::with_capacity(self.sample_types.len());
-        for sample_type in self.sample_types.iter() {
-            sample_types.push(api::ValueType {
+    fn extract_api_sample_types(&self) -> anyhow::Result<Vec<api::ValueType>> {
+        let sample_types = self
+            .sample_types
+            .iter()
+            .map(|sample_type| api::ValueType {
                 r#type: self.get_string(sample_type.r#type),
                 unit: self.get_string(sample_type.unit),
             })
-        }
-        Some(sample_types)
+            .collect();
+        Ok(sample_types)
     }
 
     /// Resets all data except the sample types and period. Returns the
     /// previous Profile on success.
-    pub fn reset(&mut self, start_time: Option<SystemTime>) -> Option<Profile> {
+    pub fn reset(&mut self, start_time: Option<SystemTime>) -> anyhow::Result<Profile> {
```

Comment:
Minor: Is it still worth returning the old profile in this situation? As an API think, it's kinda weird -- when would we want to keep around the old profile after a reset?

Reply:
I need to get this release moving so let's talk about that for v5.0. I think originally I was hoping to do  something like:

```rs
let old = profile.reset();
pprof::Profile::from(old); // owned, not borrowed
```

But that didn't work for whatever reason and forgot to revisit. Or may it's that since the structures don't match anymore (did at one time), there's not really value to take ownership of the `profile::Profile`, since it has to copy things anyway.